bigframes/core/blocks.py

```diff
@@ -2011,6 +2012,61 @@ def _is_monotonic(
         self._stats_cache[column_name].update({op_name: result})
         return result
 
+    def _get_rows_as_json_values(self) -> Block:
+        sql, index_column_ids, _ = self.to_sql_query(include_index=True)
+
+        # all column names, types and values
+        all_column_names = index_column_ids + [col for col in self.column_labels]
+        column_names_csv = ", ".join([f'"{col}"' for col in all_column_names])
+        column_references_csv = ", ".join(
+            [f"CAST({col} AS STRING)" for col in all_column_names]
+        )
+        all_column_types = list(self.index.dtypes) + list(self.dtypes)
+        column_types_csv = ", ".join([f'"{col}"' for col in all_column_types])
+
+        # index column names
+        index_column_names_csv = ", ".join([f'"{col}"' for col in index_column_ids])
+
+        row_json_column_name = guid.generate_guid()
+        select_columns = index_column_ids + [row_json_column_name]
+        select_columns_csv = ", ".join(select_columns)
+        json_sql = f"""\
+With T0 AS (
+{textwrap.indent(sql, "    ")}
+),
+T1 AS (
+    SELECT *,
+           JSON_OBJECT(
+               "names", [{column_names_csv}],
+               "types", [{column_types_csv}],
+               "values", [{column_references_csv}],
+               "index", [{index_column_names_csv}]
+           ) AS {row_json_column_name} FROM T0
```

Comment:
would it be possible to do this without generating SQL ourselves here?

Reply:
I can work on a follow up change to fit thing through our ArrayValue/nodes/ops stack (I need to understand the relationships better), would like to have something working out sooner if possible.

Reply:
Created internal issue 338218579 to track